### PR TITLE
changed structure of the exceptions

### LINKED ISSUE #28: Exception classes for topic and flashcard commands
Currently topic and flashcard commands are using exceptions that are supposed to be used by subject commands. Should have created different exceptions so that they are differentiable.

### LINKED ISSUE #48: Exception classes naming
Exception classes should have suffix "Exception", currently some exception classes under exception.card package have a different naming, and they should be renamed.

### PR DESCRIPTION
**Pull request by jialerk**
changed the Ui error messages to return Strings instead
made the catch phases of exceptions to print the exception error messages instead of from the Ui class
Added contructors to many exceptions classes
Added a few more exceptions

### CODE REVIEW

File: src/main/java/seedu/revised/exception/flashcard/NoFlashcardException.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package seedu.revised.exception.flashcard;
+
+public class NoFlashcardException extends Exception {
```
Comment (CookieHoodie) [inline src/main/java/seedu/revised/exception/flashcard/NoFlashcardException.java:3]: Inside the constructor of the classes that extends Exception, it should call super(message). In this case, it should call super(printNoFlashcardsError).

File: src/main/java/seedu/revised/ui/Ui.java [hunk 2]
```diff
@@ -101,56 +101,62 @@ public static void printBye() {
                 + "____________________________________________________________");
     }
 
-    public static void printError() {
-        System.out.println("____________________________________________________________\n"
+    public static String printFailedParseError() {
```
Comment (CookieHoodie) [inline src/main/java/seedu/revised/ui/Ui.java:104]: Personally I don't think it's a good idea to make Ui to return string the exception message in this way if no cuztomization is to be done to the message. Constants would be better in this case.

**Comment by jialerk** [comment]
closes #48 
closes #28

**Review by syncode98 — APPROVED** [review]
Edited the exception classes to ensure that the classes have a proper name.

File: src/main/java/seedu/revised/command/subject/ReturnSubjectCommand.java [hunk 3]
```diff
@@ -72,28 +72,30 @@ public void goToSubject(Subject subject) {
...
-                Ui.printNoTopicError();
+                System.out.println(e.getMessage());
             } catch (RepeatedSubjectException e) {
-                Ui.printRepeatedTopicError();
-            } catch (IndexOutOfBoundsException e) {
-                Ui.printOutOfBoundsError();
+                System.out.println(e.getMessage());
             } catch (TaskTodoException e) {
-                Ui.printTodoError();
+                System.out.println(e.getMessage());
```
Comment (CookieHoodie) [inline src/main/java/seedu/revised/command/subject/ReturnSubjectCommand.java:80]: We could create a method in Ui such as printError(Throwable err) and pass the error to that method for it to print out. In that way we let Ui handle all the printing and improves cohesion.

**Pull request merged**